lib/rubygems/specification_policy.rb

```diff
@@ -219,6 +221,15 @@ def validate_dependencies # :nodoc:
     end
   end
 
+  def validate_required_ruby_version
+    op, dep_version = @specification.required_ruby_version.requirements.first
+    segments = dep_version.segments
+
+    if (op == '>' || op == '>=') && segments == [0]
```

Comment:
I have a hard time understanding what is happening in this conditional. Does the `segments == [0]` check if the version hasn't been set yet? I wonder if moving the check into a variable with a descriptive name would make it a bit easier to follow.

Reply:
Indeed!

This is cleaner

```diff
diff --git a/lib/rubygems/specification_policy.rb b/lib/rubygems/specification_policy.rb
index a23a363e9f..b30665b9d8 100644
--- a/lib/rubygems/specification_policy.rb
+++ b/lib/rubygems/specification_policy.rb
@@ -222,10 +222,7 @@ def validate_dependencies # :nodoc:
   end
 
   def validate_required_ruby_version
-    op, dep_version = @specification.required_ruby_version.requirements.first
-    segments = dep_version.segments
-
-    if (op == '>' || op == '>=') && segments == [0]
+    if @specification.required_ruby_version.requirements == [Gem::Requirement::DefaultRequirement]
       warning "please specify minimal required ruby for your RubyGem"
     end
   end
```

The same "obscure check" is done in other validations, so we can probably extract a method with the cleaner approach and use it everywhere.

Reply:
done